### PR TITLE
chore: eslint - use parserOptions.project in order to use more eslint rules

### PR DESCRIPTION
**Pull request by TomHermanSAP**

### CODE REVIEW

**Comment by TomHermanSAP** [comment]
no-unsafe-member-access example:
image

File: .eslintrc.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ module.exports = {
     // The `ecmaVersion` should align to the supported features of our target runtimes (browsers / nodejs / others)
     // Consult with: https://kangax.github.io/compat-table/es2016plus/
     ecmaVersion: 2017,
+    project: ["./tsconfig.base.json", "./tsconfig.json"],
```
Comment (bd82) [inline .eslintrc.js:13]: should not these `parserOptions.project` be specified under the `[*.ts]` override?

File: .eslintrc.js [hunk 2]
```diff
@@ -32,6 +33,7 @@ module.exports = {
       extends: [
         "plugin:@typescript-eslint/eslint-recommended",
         "plugin:@typescript-eslint/recommended",
+        "plugin:@typescript-eslint/recommended-requiring-type-checking",
```
Comment (bd82) [inline .eslintrc.js:36]: does: `"plugin:@typescript-eslint/recommended-requiring-type-checking"`
also include "plugin:@typescript-eslint/recommended"?

**Comment by bd82** [comment]
I expended on this PR in #110 

Thanks @TomHermanSAP 👍

**Pull request closed without merging**